Remotion/ObjectBinding/Web/UI/Controls/BocBooleanValueImplementation/Rendering/BocCheckBoxRenderer.cs

```diff
@@ -222,44 +227,50 @@ private void RegisterStartupScriptIfNeeded (BocCheckBoxRenderingContext renderin
 
       string startupScript = string.Format (
           "BocCheckBox.InitializeGlobals ('{0}', '{1}');",
-          renderingContext.Control.DefaultTrueDescription,
-          renderingContext.Control.DefaultFalseDescription);
+          ScriptUtility.EscapeClientScript (renderingContext.Control.DefaultTrueDescription.ToString (WebStringEncoding.HtmlWithTransformedLineBreaks)),
+          ScriptUtility.EscapeClientScript (renderingContext.Control.DefaultFalseDescription.ToString (WebStringEncoding.HtmlWithTransformedLineBreaks)));
       renderingContext.Control.Page.ClientScript.RegisterStartupScriptBlock (renderingContext.Control, typeof (BocCheckBoxRenderer), s_startUpScriptKey, startupScript);
     }
 
-    private void PrepareImage (BocCheckBoxRenderingContext renderingContext, Image imageControl, string description)
+    private void PrepareImage (BocCheckBoxRenderingContext renderingContext, Image imageControl, WebString? description)
     {
       var imageUrl = ResourceUrlFactory.CreateThemedResourceUrl (
           typeof (HtmlHeadAppenderExtensions),
           ResourceType.Image,
           renderingContext.Control.Value.Value ? c_trueIcon : c_falseIcon);
 
       imageControl.ImageUrl = imageUrl.GetUrl();
-      imageControl.AlternateText = description ?? string.Empty;
+      imageControl.AlternateText = description?.ToString (WebStringEncoding.HtmlWithTransformedLineBreaks) ?? string.Empty;
```

Comment:
```suggestion
      imageControl.AlternateText = description?.GetValue() ?? string.Empty;
```
Sorry, I missed this before. I think this is the correct way to handle this (please check, I could miss something). AlternateText is an attribute, if we do Html or HtmlWithLinebreaks, the AlternateText would contain and then show the html characters.

Reply:
shouldn't this be `description?.ToString (WebStringEncoding.Attribute)` then?

Reply:
yeah, good question. given that asp.net already uses encoding when AlternateText is rendered, we might end up with double-encode. I don't know if there's an actual example to be found where this results in a problem, (maybe quotes?) but my gut feeling says that this would lead to problems down the line.